Userland/Libraries/LibSanitizer/UBSanitizer.cpp

```diff
@@ -46,102 +46,138 @@ static void print_location(const SourceLocation& location)
     }
 }
 
-void __ubsan_handle_load_invalid_value(const InvalidValueData&, ValueHandle) __attribute__((used));
-void __ubsan_handle_load_invalid_value(const InvalidValueData& data, ValueHandle)
+void __ubsan_handle_load_invalid_value(InvalidValueData&, ValueHandle) __attribute__((used));
+void __ubsan_handle_load_invalid_value(InvalidValueData& data, ValueHandle)
 {
+    auto location = data.location.acquire();
```

Comment:
just do the equiv of std::move(data.location) and add a move constructor for data.location

Reply:
That makes a bit more sense than what I was trying to do. I'll still keep the named function to make it clear to the reader that what we're doing is a bit more meaningful than a typical move operation.

Reply:
Sadly the compiler generated move operations didn't clear out the source, So I had to implement them manually... :man_shrugging: